src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/README.md

```diff
@@ -0,0 +1,26 @@
+# Common Language Server Protocol Framework (CLaSP)
+
+CLaSP was created to ease the creation and maintenance of [Language Server](https://microsoft.github.io/language-server-protocol/) implementations by sharing our hard-won knowledge, while leaving you the flexiblity to create the server you need.
+
+## Getting Started with CLaSP
+
+You can find a [simple example implementation of a CLaSP-based LanguageServer here](https://github.com/dotnet/roslyn/tree/main/src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework.Example).
+
+To get started with CLaSP you may follow the following steps
+
+1. Create a PackageReference to `Microsoft.CommonLanguageServerProtocol.Framework` in your LSP server project.
+1. Implement the following classes:
+    1. `ILspLogger`. This allows CLaSP to log information however you would like.
+    1. `ILspServices`. This interface will serve as a wrapper around whatever DI system you choose to use to make sure that your services implemented elsewhere are available to CLaSP.
+    1. `AbstractLanguageServer`. This is the core of your Language Server implementation, which will manage your lifecycle and host the rest of the components.
+        1. Ensure that `Initialize` is called from the constructor to start the server.
```

Comment:
I think Initialize could be called whenever right?  It just has to be called before the server starts accepting requests.